api/base/serializers.py

```diff
@@ -1837,3 +1839,27 @@ class MaintenanceStateSerializer(ser.ModelSerializer):
     class Meta:
         model = MaintenanceState
         fields = ('level', 'message', 'start', 'end')
+
+
+class HideIfSwitch(ConditionalField):
+    """
+    If switched is switched this field is hidden.
+    """
+    def __init__(self, switch_name, field, hide_if=False, **kwargs):
+        super(HideIfSwitch, self).__init__(field, **kwargs)
+        self.switch_name = switch_name
+        self.hide_if = hide_if
+
+    def should_hide(self, instance):
+        return waffle.switch_is_active(self.switch_name) == self.hide_if
+
+
+class DisableIfSwitch(HideIfSwitch):
+    """
+    If switched is switched this field is hidden.
```

Comment:
This docstring should be different from the class it descends from to indicate what the difference is.